### ISSUE TITLE
TypedTableBlocks do not display errors to users when part of a StructBlock

### ISSUE THREAD

**Issue by haydngreatnews** [wagtail/wagtail#7997]
### Issue Summary

When adding a TypedTableBlock to a StructBlock, I wanted to add some extra validation to restrict the table's width.
I added a `clean` method to the StructBlock, and raised a StreamBlockValidationError within it, to set the error for the table block. This error, although I can see it in the page source is never rendered to the page, and so does not give any user feedback.

### Steps to Reproduce

1. (for example) Start a new project with `wagtail start myproject`
2. Add a StructBlock with a definition like
```python
class DecoratedTableBlock(blocks.StructBlock):
    table = TypedTableBlock(
        [
            ("text", blocks.TextBlock()),        ],
    )
    caption = ReducedRichTextBlock(required=False)

    def clean(self, value):
        value = super().clean(value)
        table = value.get("table")
        raise blocks.StreamBlockValidationError(
            non_block_errors=ErrorList(["Validation error in table"]),
            block_errors={
                "caption": ErrorList(
                    ["Error message for table"]
                )
            },
        )
        return value
```
3. Add the StructBlock to a Page model's StreamField
4. Add `wagtail.contrib.typed_table_block` to `INSTALLED_APPS`
4. Run migrations
5. Add an instance of the StructBlock via the CMS
6. Save the page draft
1. This will fail and the red ribbon banner will pop up
2. There will be no error highlighting or text on the table block

I would expect that raising an error in this manner would decorate the child block with appropriate error decorations, and display the text defined in the ErrorList.

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes - See [gist for full models](https://gist.github.com/haydnnewport/8177de5e78ca1feeddd888c574ddada9) file

### Technical details

* Python version: 3.9,4.
* Django version: 3.2.12
* Wagtail version: 2.16.1
* Browser version: Chrome 97 - https://www.whatsmybrowser.org/b/C6EUV

**Comment by gasman** [comment]
This will be fixed by #10131.

**Comment by gasman** [comment]
Fixed in #10131
